src/main/java/uk/nhs/tis/trainee/notifications/model/NotificationScheduleWhen.java

```diff
@@ -0,0 +1,12 @@
+package uk.nhs.tis.trainee.notifications.model;
+
+import java.util.Date;
+
+/**
+ * A record representing the scheduling of a notification.
+ *
+ * @param unnecessaryReminder Indicates if the notification is an unnecessary reminder.
+ * @param scheduleWhen        The date when the notification is scheduled to be sent.
+ */
+public record NotificationScheduleWhen(boolean unnecessaryReminder, Date scheduleWhen) {
```

Comment:
This is a fudge to allow `ProgrammeMembershipService.whenScheduleProgrammeNotification()` to indicate if the notification should _not_ be sent at all, as in the case of an overdue reminder notification.

Reply:
Ok, got rid of this by simply not scheduling/logging overdue or unneeded reminders